src/transformers/audio_utils.py

```diff
@@ -81,279 +76,521 @@ def mel_to_hertz(mels: np.array, mel_scale: str = "htk") -> np.array:
     if mel_scale == "htk":
         return 700.0 * (10.0 ** (mels / 2595.0) - 1.0)
 
-    # Fill in the linear scale
-    frequency_min = 0.0
-    f_sp = 200.0 / 3
-    freqs = frequency_min + f_sp * mels
-
-    # And now the nonlinear scale
     min_log_hertz = 1000.0
-    min_log_mel = (min_log_hertz - frequency_min) / f_sp
-    logstep = math.log(6.4) / 27.0
+    min_log_mel = 15.0
+    logstep = np.log(6.4) / 27.0
+    freq = 200.0 * mels / 3.0
 
-    log_t = mels >= min_log_mel
-    freqs[log_t] = min_log_hertz * np.exp(logstep * (mels[log_t] - min_log_mel))
+    if isinstance(mels, np.ndarray):
+        log_region = mels >= min_log_mel
+        freq[log_region] = min_log_hertz * np.exp(logstep * (mels[log_region] - min_log_mel))
+    elif mels >= min_log_mel:
+        freq = min_log_hertz * np.exp(logstep * (mels - min_log_mel))
 
-    return freqs
+    return freq
 
 
-def _create_triangular_filterbank(
-    all_freqs: np.array,
-    f_pts: np.array,
-) -> np.array:
-    """Create a triangular filter bank.
+def _create_triangular_filter_bank(fft_freqs: np.ndarray, filter_freqs: np.ndarray) -> np.ndarray:
+    """
+    Creates a triangular filter bank.
 
+    Adapted from *torchaudio* and *librosa*.
 
     Args:
-        all_freqs (`np.array` of shape (`nb_frequency_bins`, )):
-            Discrete frequencies used when the STFT was computed.
-        f_pts (`np.array`, of shape (`nb_mel_filters`, )):
-            Coordinates of the middle points of the triangular filters to create.
+        fft_freqs (`np.ndarray` of shape `(num_frequency_bins,)`):
+            Discrete frequencies of the FFT bins in Hz.
+        filter_freqs (`np.ndarray` of shape `(num_mel_filters,)`):
+            Center frequencies of the triangular filters to create, in Hz.
 
     Returns:
-        fb (np.array):
-            The filter bank of size (`nb_frequency_bins`, `nb_mel_filters`).
+        `np.ndarray` of shape `(num_frequency_bins, num_mel_filters)`
     """
-    # Adapted from Librosa
-    # calculate the difference between each filter mid point and each stft freq point in hertz
-    f_diff = f_pts[1:] - f_pts[:-1]  # (n_filter + 1)
-    slopes = np.expand_dims(f_pts, 0) - np.expand_dims(all_freqs, 1)  # (nb_frequency_bins, n_filter + 2)
-    # create overlapping triangles
-    zero = np.zeros(1)
-    down_slopes = (-1.0 * slopes[:, :-2]) / f_diff[:-1]  # (nb_frequency_bins, n_filter)
-    up_slopes = slopes[:, 2:] / f_diff[1:]  # (nb_frequency_bins, n_filter)
-    fb = np.maximum(zero, np.minimum(down_slopes, up_slopes))
-
-    return fb
-
-
-def get_mel_filter_banks(
```

Comment:
Can't rename a documented function without at least a deprecation cycle. Since those are utils, let's maybe avoid all of this and not do the rename?